### PR TITLE
Consolidate changes by affected packages

### PR DESCRIPTION
**Pull request by gigabo**
This addresses #16.

The diff wound up not so clean here, unfortunately.  I also made some changes to to how the list of affected packages is generated.  These may be more legible in their logical commits.

### CODE REVIEW

File: src/Changelog.js [hunk 1]
```diff
@@ -40,60 +40,57 @@ export default class Changelog {
...
+            if (files.indexOf("packages/") === 0) {
+              obj[files.slice(9).split("/", 1)[0]] = true;
+            }
+            return obj;
+          }, {})
+        );
+
+        const heading = changedPackages.length > 0
+          ?"* "+changedPackages.map(pkg => "`" + pkg + "`").join(", ")
+          :"* Other"; // No changes to packages, but still relevant.
```
Comment (gigabo) [inline src/Changelog.js]: This is worth mentioning: Previously changes with no associated packages bumped out to the outer list level.  This branch moves them into an "Other" group.  These are changes that get assigned to a changelog category but don't affect any packages.

Comment (hzoo) [inline src/Changelog.js]: Good point I just choose to ignore those (top level changes) or manually add it in but it makes sense to have

**Comment by gigabo** [comment]
@hzoo - This look okay to you?

**Comment by hzoo** [comment]
hard to say haha - if we had a test or two this would be nice. probably fine. maybe we can split up logic into multiple simpler functions

**Comment by gigabo** [comment]
Yeah, I'd like to set up tests with repo fixtures [like @thejameskyle suggested](https://github.com/lerna/lerna-changelog/issues/19#issuecomment-222847111).  I'd like to use some helpers from lerna's tests ([_initFixture](https://github.com/lerna/lerna/blob/master/test/_initFixture.js), [_assertStubbedCalls](https://github.com/lerna/lerna/blob/master/test/_assertStubbedCalls.js), ...), but it's a shame to just duplicate them.

I [half-joked on slack](https://lernajs.slack.com/archives/discussion/p1464797305000031) that lerna should be a lerna repo with `lerna` and `lerna-changelog` as packages, but that would actually help with sharing stuff like test helpers between projects.  It would also allow using `lerna-changelog` to generate a changelog for lerna itself.

Anyway, yes: I'll split this up into smaller unit-testable functions.

**Comment by hzoo** [comment]
We can go ahead a merge given it's just us using it atm and we can make quick changes

**Pull request merged**